haystack/utils/device.py

```diff
@@ -389,7 +389,7 @@ def has_multiple_devices(self) -> bool:
         return self._multiple_devices is not None
 
     @property
-    def first_device(self) -> Optional[Device]:
```

Comment:
As a heads up I removed the Optional here since I don't believe it is actually needed and causes mypy errors when trying to use `first_device.to_torch()`

Reply:
Hmm, I think we still need to keep it? Currently, there is no guarantee that `DeviceMap` won't be empty, so the assertion on line 410 can get triggered easily. If we want to avoid the optional here - and there is a valid argument to be made about the semantics - I'd sooner modify the `ComponentDevice.from_multiple` to raise an error if the passed device map is empty. Then, it becomes a reliable invariant.

PS: The `mypy` error can be "fixed" by with `assert first_device is not None` at the callsite (or with a `# type: ignore`, if you so choose).